### DISCUSSION TITLE
Is there's a tool or reference for converting SQL to DAX?

### DISCUSSION THREAD

**Discussion by UncleDreiAI** [DaxStudio/DaxStudio#1151] [Q&A] [answered]
Hello, I am leveling up my DAX, and coming from the SQL world. While I am wrapping my head around the language, I was wondering if there are any references to guide the SQL to DAX users?

**Comment by marcosqlbi** [comment, marked as answer]
I wrote a series of article "From SQL to DAX" - you can see the index at the bottom of each article, this is the first one:
https://www.sqlbi.com/articles/from-sql-to-dax-projection/

However, please note that learning DAX by example does not work well if you want to be really proficient with DAX, as you should learn new concepts (filter context, row context, context transition) that don't exist in SQL and can confuse you. This is a guide to learn DAX starting with free resources:
https://www.sqlbi.com/guides/dax/

If you use DAX mainly to query a model like you would do in SQL and not to create measures, we also have a specific course designed with this purpose (you should learn the DAX basics before):
https://www.sqlbi.com/whitepapers/writing-dax-queries/

**Reply by UncleDreiAI** [reply]
Awesome! Thanks for all that you do!
